mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td

```diff
@@ -163,7 +174,26 @@ def XeGPU_CreateNdDescOp: XeGPU_Op<"create_nd_tdesc", [Pure, ViewLikeOpInterface
     }
 
     ArrayRef<int64_t> getStaticOffsets(){
-      return getConstOffsets();
+      auto attr = getConstOffsetsAttr();
+
+      if (attr) 
+        return attr;
+
+      auto memrefType = llvm::dyn_cast<MemRefType>(getSourceType());
+      int rank = 0;
+      if (memrefType) 
+        //use source memref's rank, as source memref rank may be higher
+        rank = memrefType.getRank();
+      else
+        //nd_tdesc created from ui64, use nd_tdesc's rank
+        rank = getMixedSizes().size();
+      
+      // The offsets are allowed to be empty. The Traits verification of OffsetSizeAndStrideOpInterface interface assumes offsets being present.
+      // place-holder value MAX indicating user doesn't provide offsets, instead of kDynamic which means offsets passed as value.
+      setConstOffsets(llvm::SmallVector<int64_t, 4>(rank, std::numeric_limits<int64_t>::max()));
```

Comment:
Why is the op property being modified in a getter? This field should be populated correctly at the op construction.

Reply:
> place-holder value MAX indicating user doesn't provide offsets

I don't think the interface understands that which breaks the whole abstraction.
Can offsets default to zero when they're not provided?

Reply:
This is also related to OffsetSizeAndStrideOpInterface  issue we discussed below